util/refobject_base.h

```diff
@@ -51,16 +51,17 @@ namespace leveldb {
  */
 class RefObjectBase
 {
- protected:
+    // force this private so everyone is using memory fenced GetRefCount
+ private:
     volatile uint32_t m_RefCount;
 
  public:
     RefObjectBase() : m_RefCount(0) {}
     virtual ~RefObjectBase() {}
 
-    uint32_t RefInc() {return(inc_and_fetch(&m_RefCount));}
+    virtual uint32_t RefInc() {return(inc_and_fetch(&m_RefCount));}
 
-    uint32_t RefDec()
+    virtual uint32_t RefDec()
```

Comment:
Since ErlRefObject overrides RefDec(), I wonder if not having RefObjectBase::RefDec() be virtual could have been a problem prior to this change?